### PR TITLE
Feat/motion monitor

### PR DESCRIPTION
**Pull request by Tdreamy**
- add motion monitor

### CODE REVIEW

File: iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//
+//  Created by tianYang on 2022/11/4.
+//
+
+#import "DoraemonMotionDataSource.h"
+
+@implementation DoraemonMotionDataSource {
+    dispatch_semaphore_t semaphore;
+}
++ (DoraemonMotionDataSource *)shareInstance{
```
Comment (tomisacat) [inline iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m:13]: `sharedInstance` or `shared` looks like a better name?

File: iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+- (void)clear{
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    [_motionUseModelArray removeAllObjects];
+    dispatch_semaphore_signal(semaphore);
+}
+
+- (NSString *)toJson {
+    NSMutableArray *dicArray = @[].mutableCopy;
```
Comment (tomisacat) [inline iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m:44]: why not surrounded by semaphore here?

File: iOS/DoraemonKit/Src/Core/Util/Aspects/Aspects.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+//  Aspects.h
+//  Aspects - A delightful, simple library for aspect oriented programming.
+//
+//  Copyright (c) 2014 Peter Steinberger. Licensed under the MIT license.
+//
+
+#import <Foundation/Foundation.h>
+
+typedef NS_OPTIONS(NSUInteger, AspectOptions) {
+    AspectPositionAfter   = 0,            /// Called after the original implementation (default)
```
Comment (tomisacat) [inline iOS/DoraemonKit/Src/Core/Util/Aspects/Aspects.h:11]: `AspectOption*` instead of `AspectPosition*`?

**Review by tomisacat — COMMENTED** [review]
some minor changes, no impact on merging.

**Comment by jhonny-me** [comment]
Good Job!

The next step would be how to analyze these data and generate reports.

**Review by jhonny-me — APPROVED** [review]

File: iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m (hunk 2 above)
Comment (Tdreamy) [reply iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m:44]: We use the same naming method as the other singletons in Dokit
image

File: iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m (hunk 1 above)
Comment (Tdreamy) [reply iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m:13]: We use the same naming method as the other singletons in Dokit
image

File: iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m (hunk 2 above)
Comment (Tdreamy) [reply iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m:44]: in here，we just read the _motionUseModelArray

File: iOS/DoraemonKit/Src/Core/Util/Aspects/Aspects.h (hunk 3 above)
Comment (Tdreamy) [reply iOS/DoraemonKit/Src/Core/Util/Aspects/Aspects.h:11]: A little confused, this is a third party library code, we'd better keep the original code

**Pull request merged**